fri/src/verifier/channel.rs

```diff
@@ -90,26 +90,17 @@ pub trait VerifierChannel<E: FieldElement> {
         Ok(group_vector_elements(layer_queries))
     }
 
-    /// Returns FRI remainder values (last FRI layer) read from this channel.
+    /// Returns FRI remainder polynomial read from this channel.
     ///
-    /// This also checks whether the remainder is valid against the provided commitment.
+    /// This also checks whether the remainder is valid against the provided maximum expected degree.
     ///
     /// # Errors
     /// Returns an error if:
-    /// - If the sequrntial hash constructed from the remainder values does not match the specified
-    ///   `commitment`.
-    fn read_remainder<const N: usize>(
-        &mut self,
-        commitment: &<<Self as VerifierChannel<E>>::Hasher as Hasher>::Digest,
-    ) -> Result<Vec<E>, VerifierError> {
+    /// - If the degree of the polynomial is greater than the maximum expected degree.
+    fn read_remainder(&mut self, max_degree: usize) -> Result<Vec<E>, VerifierError> {
         let remainder = self.take_fri_remainder();
-
-        // sequential hash of the remainder
-        let set_commitment =
-            <<Self as VerifierChannel<E>>::Hasher as ElementHasher>::hash_elements(&remainder);
-
-        if *commitment != set_commitment {
-            return Err(VerifierError::RemainderCommitmentMismatch);
+        if polynom::degree_of(&remainder) > max_degree {
+            return Err(VerifierError::RemainderDegreeMismatch(max_degree));
         }
```

Comment:
I think it would be cleaner to perform this check in the verifier (i.e., in `verify_generic()` function).

Also, can we simplify this as:
```Rust
if remainder.len() > max_degree + 1 {
    return Err(VerifierError::RemainderDegreeMismatch(max_degree));
}
```

Reply:
Done!